homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,390 @@
+"""Platform for eQ-3 climate entities."""
+
+import asyncio
+from collections.abc import Callable
+from datetime import datetime, timedelta
+import logging
+from typing import Any
+
+from eq3btsmart import Thermostat
+from eq3btsmart.const import EQ3BT_MAX_TEMP, EQ3BT_OFF_TEMP, Eq3Preset, OperationMode
+from eq3btsmart.exceptions import Eq3Exception
+
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_TENTHS, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.device_registry import (
+    CONNECTION_BLUETOOTH,
+    DeviceInfo,
+    async_get,
+    format_mac,
+)
+from homeassistant.helpers.entity import Entity, EntityPlatformState
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+
+from .const import (
+    DEVICE_MODEL,
+    DOMAIN,
+    EQ_TO_HA_HVAC,
+    HA_TO_EQ_HVAC,
+    MANUFACTURER,
+    CurrentTemperatureSelector,
+    Preset,
+    TargetTemperatureSelector,
+)
+from .eq3_entity import Eq3Entity
+from .models import Eq3Config, Eq3ConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Handle config entry setup."""
+
+    eq3_config_entry: Eq3ConfigEntry = hass.data[DOMAIN][config_entry.entry_id]
+    thermostat = eq3_config_entry.thermostat
+    eq3_config = eq3_config_entry.eq3_config
+
+    entities_to_add: list[Entity] = [Eq3Climate(eq3_config, thermostat)]
+
+    async_add_entities(
+        entities_to_add,
+        update_before_add=False,
+    )
```

Comment:
```suggestion
    async_add_entities(
        [Eq3Climate(eq3_config_entry.eq3_config, eq3_config_entry.thermostat,)]
    )
```

Reply:
I normally like to be a bit more expressive and have this in separate lines, especially since in the platforms that will be added in the future some entities are only created based on a flag set in the options flow, but I guess I'll simplify this in the way you suggested.